src/test/ui/privacy/issue-46209-private-enum-variant-reexport.rs

```diff
@@ -1,15 +1,16 @@
 #![feature(crate_visibility_modifier)]
 
+#[deny(unused_imports)]
 mod rank {
     pub use self::Professor::*;
-    //~^ ERROR enum is private and its variants cannot be re-exported
+    //~^ ERROR glob import doesn't reexport anything
```

Comment:
Is this a change in behaviour, or is the lint taking precedence over the "enum is private" error?

Reply:
This is a change in behavior, now glob import from enum with private variants behaves identically to a glob import from a module with private items.
Previously, in the world without `pub(restricted)`, this required special logic and separate error.

Reply:
This definitely seems like better behaviour, but is this something that would need to undergo a lang team FCP, or is this a change that has already been discussed?

Reply:
Maybe.
When the error was added in https://github.com/rust-lang/rust/pull/29973/commits/8f359d5912de9162534d65fe01fb2f52941e97d0 (as a part of hardening private-in-public errors), the glob case wasn't really discussed.

The glob error was added pretty conservatively to seal any possible sources of leakage, but even them it was inconsistent with behavior of other glob imports.

Reply:
Let's cc them just to make sure, though it's likely this isn't large enough to warrant an FCP.